jobsdb/jobsdb.go

```diff
@@ -3007,7 +3003,28 @@ func (jd *HandleT) backupTable(backupDSRange dataSetRangeT, isJobStatusTable boo
 	return true, nil
 }
 
-func (jd *HandleT) getBackupDSRange() dataSetRangeT {
+func (jd *HandleT) backupUploadWithExponentialBackoff(ctx context.Context, file *os.File, pathPrefixes ...string) (filemanager.UploadOutput, error) {
+	// get a file uploader
+	fileUploader, errored := jd.getFileUploader()
+	jd.assertError(errored)
+
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = time.Minute
+	bo.MaxElapsedTime = jd.maxBackupRetryTime
+	boCtx := backoff.WithContext(bo, ctx)
+
+	var err error
+	var output filemanager.UploadOutput
+	backup := func() error {
```

Comment:
minor: for simplicity you can consider using an anonymous function, e.g. 
```go
var err error
var output filemanager.UploadOutput
err = backoff.Retry(func() error {
    output, err = fileUploader.Upload(ctx, file, pathPrefixes...)
    return err
}, boCtx)
return output, err
```

Reply:
@chrikar pointed out last time that, this decreases the code readability. So, did it like this here. 
ref: https://github.com/rudderlabs/rudder-server/pull/1770#discussion_r837354087

Reply:
Yeap, that's why I mentioned this is minor, .i.e. subjective. What is readable to me might not be for someone else!